kernel/sched.c

```diff
@@ -522,19 +526,15 @@ void z_time_slice(int ticks)
 #endif
 
 	if (slice_time(_current) && sliceable(_current)) {
-		if (ticks >= _current_cpu->slice_ticks) {
+		if (slice_expired[_current_cpu->id]) {
 			/* Note: this will (if so enabled) internally
 			 * drop and reacquire the scheduler lock
 			 * around the callback!  Don't put anything
 			 * after this line that requires
 			 * synchronization.
 			 */
 			key = slice_expired_locked(key);
-		} else {
-			_current_cpu->slice_ticks -= ticks;
 		}
-	} else {
-		_current_cpu->slice_ticks = 0;
```

Comment:
Look at all the logic that goes away when the tick accounting happens just once at reset time